### PR TITLE
Add DAG file processing description to Scheduler Concepts

### PR DESCRIPTION
**Pull request by john-jac**
One of the challenges for Airflow users it to understand how the Airflow Scheduler turns Python code into DAGs.  Often users will be confused as to why their DAGs don't show up in the user interface, or they'll see DAG processing errors and not know why.

This documentation entry is intended to clarify the process to new and experienced users alike, by providing a concise description of the process, along with a diagram of the flow and links to key configuration values that affect the way Airflow processes DAGs.

### CODE REVIEW

**Comment by uranusjr** [comment]
The content looks very good to me. The only concern I have is, this is going into Airflow internals that don't seem to fit with other things under *Concept* (which are mostly about user-facing features). But I don't really have a good suggestion where else this should go.

**Comment by john-jac** [comment]
> The content looks very good to me. The only concern I have is, this is going into Airflow internals that don't seem to fit with other things under _Concept_ (which are mostly about user-facing features). But I don't really have a good suggestion where else this should go.

I tried to align the format and content with https://airflow.apache.org/docs/apache-airflow/stable/concepts/tasks.html which is in _Concepts_ as well so it probably fits as well as anywhere.

File: docs/apache-airflow/concepts/scheduler.rst [hunk 1]
```diff
@@ -60,6 +60,34 @@ In the UI, it appears as if Airflow is running your tasks a day **late**
 
     You should refer to :doc:`/dag-run` for details on scheduling a DAG.
 
+DAG File Processing
+-------------------
+
+The Airflow Scheduler is responsible for turning the Python files contained in the DAGs folder into DAG objects that contain tasks to be scheduled.
+
+There are two primary components involved in DAG file processing.  The ``DagFileProcessorManager`` is an infinite loop that determines which files need
```
Comment (ashb) [inline docs/apache-airflow/concepts/scheduler.rst]: I think my only comment here is the Manager also runs in a separate process to the scheduler.

(Sorry for the delay in reviewing this @john-jac!)

Comment (john-jac) [reply docs/apache-airflow/concepts/scheduler.rst]: No worries @ashb !  I've updated the doc with that distinction.

**Review by ashb — APPROVED** [review]

**Pull request merged**